### PR TITLE
remove assumption that merged and fixed variables are disjoint 

### PR DESCRIPTION
**Pull request by yuvaljacoby**
It's a minor change.
In Preprocessor.cpp we assume that fixedVariables and mergedVariables are disjoint (preprocess function).

I tried an example that it wasn't the case and then number of variables overflowed and the query got stuck. 
Not the most elegant solution but fixed my problem...

### CODE REVIEW

**Comment by guykatzz** [comment]
I don't really understand this fix. You're suggesting that the "variableCount" can be negative, and that it should be set to 0 instead? We should add a more general fix, as it seems this phenomenon can happen also when variableCount is positive.

**Comment by yuvaljacoby** [comment]
You are probably right.
This fix is a patch and not what we actually need to do.
I pushed code that reproduce the problem, I prefer not to fix it right now (after finishing CMake I can do it). In the mean time should I also open an issue?

File: src/engine/Makefile [hunk 1]
```diff
@@ -44,13 +44,8 @@ vpath %.cpp $(ENGINE_REAL_DIR)
...
 
 SOURCES += \
-	DnCMarabou.cpp \
-	Marabou.cpp \
-	OptionParser.cpp \
-	Options.cpp \
-	main.cpp \
-
-TARGET = marabou.elf
+		   variables_test.cpp
```
Comment (guykatzz) [inline src/engine/Makefile]: I think this cpp file is missing

**Comment by guykatzz** [comment]
For the failing example, did you run in DEBUG mode? Did the assertion at the end of Preprocessor::preprocess go off?

**Comment by yuvaljacoby** [comment]
Didn't run in debug mode, didn't know about the DEBUG_ON flag.
After adding it the assert raises exception.

**Comment by guykatzz** [comment]
Please try now

**Comment by yuvaljacoby** [comment]
Works, Thanks.
Cleaned the pull request, I think we can squash and merge.

**Comment by guykatzz** [comment]
Great.
Before we merge, please add your previously failing example as either a
unit test for Preprocessor.h, or as a regression test, and see that it
still fails if you comment out the new function call.

**Comment by yuvaljacoby** [comment]
done 
did also a small fix in sec/common/debug , we need the three dots there in order to use variables.

The default compilation of the tests does not have the DEBUG_ON flag, should we change it?

**Comment by guykatzz** [comment]
Yes, lets have DBEUG_ON for the tests.

**Pull request merged**